### ISSUE TITLE
Missing support for .chai files

### ISSUE THREAD

**Issue by Ybalrid** [madskristensen/TextmateBundleInstaller#75]
Okay, this is great, I just commented on your page for the extension in the Visual Studio site, and after the update, your extension propose to open issues here automatically ^^"

So, ".chai" files are script files for ChaiScript. ChaiScript ins an embedded scripting language that is made to be super easy to integrate in a C++ application. 
It's a project made by @lefticus (@ChaiScript organisation) and the homepage is here : 

http://chaiscript.com/

There's an example of how this thing looks and works here http://chaiscript.com/examples.html#ChaiScript_Language_Examples

It would be super neat to have this works with Visual Studio. I don't think there's a "Textmate bundle" for it, I'm afraid.

(I'm willing to take some time to help, I just have to look into how this works... :wink:)

**Comment by madskristensen** [comment]
I can't seem to find the textmate bundle for ChaiScript, unfortunately.

**Comment by Ybalrid** [comment]
There is not, but should be doable to make one... 

I don't know textmate, and I don't have a mac to try it. But looking at the files in your repo, theses bundle look like to be just some XML-ish files. 

Making something that highlight a bunch of keywords "def, class, var, attr, auto..", strings and literals should not be super hard (I hope). 

There's the list of keywords in the headers of ChaiScript and in the syntax highlighting files for Notepad++ or Vim that can also be helpful to build that.

**Comment by lefticus** [comment]
I suggest that for anyone wanting to do this that the ChaiScript vim file is the most up-to-date example for syntax support: https://github.com/ChaiScript/vim-chaiscript/blob/master/syntax/chaiscript.vim

**Comment by Ybalrid** [comment]
@lefticus Thank you, I'll try to make something work with that.

I still have to understand how textmates bundles actually works and how I can try out what I'm doing without breaking anything 

@madskristensen how do you test those on something other than Textmate. I don't have a mac and I have no experience with Visual Studio extension development, so I don't know how I could try a "bundle" with your extension easily ^^"

**Comment by madskristensen** [comment]
The textmate support in VS doesn't require any extensibility knowhow. All this extension does is adding textmate bundles to `%userprofile%\.vs\Extensions`. Visual Studio natively picks reads from that folder. You can do the same manually (you may have to disable this extension while manually working in the `%userprofile%\.vs\Extensions` folder).

You may have to restart VS to test the changes you've made in the folder or at least close and reopen the file in the editor - I'm not sure about that.

**Comment by Ybalrid** [comment]
Oh, so it just works like this? Mh... Looks like I have everything on hand to make that happen then... 😛

**Comment by madskristensen** [comment]
Yeah, it's pretty cool. Unfortunately I've never written a textmate bundle, so I can't help out much on that aspect

**Comment by dean1012** [comment]
@madskristensen @lefticus While it isn't a perfect solution, it seems that Chaiscript is fairly similar to Javascript. Can we just associate .chai with .js to have the Javascript highlighting applied at least until something better comes along?
